_posts/2023-08-29-what-in-the-ml-is-going-on-around-here.md

```diff
@@ -0,0 +1,473 @@
+
+The error message was confusing, so I filed [issue 1289](https://github.com/opensearch-project/ml-commons/issues/1289).
+
+
+### Side Quest: Tasks, Models, and Model Groups
+
+Many of the API operations we'll end up using return a task id, model id, or model group id that you have to then look up. Since we're only using the tools that come out of the box, here are the dev console examples that are likely of most help.
+
+```json
+# To see what models exist.
+GET /_plugins/_ml/models/_search
+{ "query": { "match_all": {} } }
+
+# To delete a model
+DELETE /_plugins/_ml/models/`model_id`
+
+# To see the model groups that exist. 
+GET /_plugins/_ml/model_groups/_search
+{ "query": { "match_all": {} } }
+
+# To delete a model group
+DELETE /_plugins/_ml/model_groups/`model_group_id`
+
+# To see the tasks that exist.
+GET /_plugins/_ml/tasks/_search
+{ "query": { "match_all": {} } }
+
+# To delete a task
+DELETE /_plugins/_ml/tasks/`task_id`
+```
+
+----
+
+Something serendipitous happened as I was trying to teach myself the API for ml-commons when following my tasks, models, and model groups side quest. I found an API call example
+[here](https://opensearch.org/docs/latest/ml-commons-plugin/api/) on the ml-commons api reference page. It was perfectly formed, and included all of the stuff that I was missing. Here it is: 
+
+```json
+POST /_plugins/_ml/models/_register
+{
+    "name": "all-MiniLM-L6-v2",
+    "version": "1.0.0",
+    "description": "test model",
+    "model_format": "TORCH_SCRIPT",
+    "model_group_id": "FTNlQ4gBYW0Qyy5ZoxfR",
+    "model_content_hash_value": "c15f0d2e62d872be5b5bc6c84d2e0f4921541e29fefbef51d59cc10a8ae30e0f",
+    "model_config": {
+        "model_type": "bert",
+        "embedding_dimension": 384,
+        "framework_type": "sentence_transformers",
+       "all_config": "{\"_name_or_path\":\"nreimers/MiniLM-L6-H384-uncased\",\"architectures\":[\"BertModel\"],\"attention_probs_dropout_prob\":0.1,\"gradient_checkpointing\":false,\"hidden_act\":\"gelu\",\"hidden_dropout_prob\":0.1,\"hidden_size\":384,\"initializer_range\":0.02,\"intermediate_size\":1536,\"layer_norm_eps\":1e-12,\"max_position_embeddings\":512,\"model_type\":\"bert\",\"num_attention_heads\":12,\"num_hidden_layers\":6,\"pad_token_id\":0,\"position_embedding_type\":\"absolute\",\"transformers_version\":\"4.8.2\",\"type_vocab_size\":2,\"use_cache\":true,\"vocab_size\":30522}"
+    },
+    "url": "https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-transformers/all-MiniLM-L6-v2/1.0.1/torch_script/sentence-transformers_all-MiniLM-L6-v2-1.0.1-torch_script.zip"
+}
+```
+
+It still begged of me the question, "How do I know what values to put under model_config if I'm using some other model?" Then I came across *another* example call referring to something called a "Pretrained model" - look at this call: 
```

Comment:
This is an interesting one because it's an example of language that is currently in the process of changing. According to Merriam-Webster, begging the question means "to elicit a specific question as a reaction or response," and can often be replaced with "a question that begs to be answered." However, a lesser used and more formal definition is "to ignore a question under the assumption it has already been answered." The phrase itself comes from a translation of an Aristotelian phrase rendered as "beg the question" but meaning "assume the conclusion." In other words, "beg the question" has not historically meant "raise the question," but it's been used in that way by so many people for so long that its use is becoming acceptable. I still think "raised" just sounds better here 😄.